### PR TITLE
add --source-exclude and --message-exclude, --id

### PR DESCRIPTION
**Pull request by daiksy**
for check-windows-eventlog

### CODE REVIEW

File: check-windows-eventlog/README.md [hunk 1]
```diff
@@ -25,3 +25,23 @@ command = "/path/to/check-windows-eventlog --log=LOGTYPE --type=EVENTTYPE --sour
...
+1. find message matches `foo` but not match `bar`.
+
+    ```
+    --message-pattern foo --message-exclude bar
+    ```
+
+2. find event which id is 900 or 901.
+
+    ```
+    --id 900,901
```
Comment (itchyny) [inline check-windows-eventlog/README.md]: IMO, `--event-id` is preferable...

Comment (itchyny) [inline check-windows-eventlog/README.md]: Is there any specific reason that `--id` is more suitable for this option?

File: check-windows-eventlog/check-windows-eventlog.go [hunk 2]
```diff
@@ -28,11 +28,25 @@ const (
...
+}
+
 type logOpts struct {
 	Log            string `long:"log" description:"Event Names (comma separated)"`
 	Type           string `long:"type" description:"Event Types (comma separated)"`
 	SourcePattern  string `long:"source-pattern" description:"Event Source (regexp pattern)"`
+	SourceExclude  string `long:"source-exclude" description:"Event Source excluded (regexp pattern)"`
 	MessagePattern string `long:"message-pattern" description:"Message Pattern (regexp pattern)"`
+	MessageExclude string `long:"message-exclude" description:"Message Pattern excluded (regexp pattern)"`
+	EventID        string `long:"event-id" description:"Event IDs (separated by comma)"`
```
Comment (astj) [inline check-windows-eventlog/check-windows-eventlog.go]: Not necessary, but it's better if range specification like `${id1}-${id2}` and excluding by `!${id}` are described.

File: check-windows-eventlog/check-windows-eventlog.go [hunk 3]
```diff
@@ -293,6 +371,30 @@ func (opts *logOpts) searchLog(logName string) (warnNum, critNum int64, errLines
...
+				if !idr.bang {
+					exact = true
+					if idr.lo <= eventID && eventID <= idr.hi {
+						found = true
+					}
+				} else {
+					if idr.lo <= eventID && eventID <= idr.hi {
+						found = false
+					} else if !exact {
+						found = true
```
Comment (astj) [inline check-windows-eventlog/check-windows-eventlog.go]: `!2,!3` against `2` maybe become `found = true`, though `!3,!2` against `2` maybe become `found = false`. Is this behavior accepted? (imo acceptable...)

File: check-windows-eventlog/README.md (hunk 1 above)
Comment (mattn) [inline check-windows-eventlog/README.md]: thanks for your review.
make sense, it looks better.

File: check-windows-eventlog/check-windows-eventlog.go (hunk 3 above)
Comment (mattn) [inline check-windows-eventlog/check-windows-eventlog.go]: thanks for your review.
hmm, this code is confusible. :(

|No|Seq  |Exact|Found|
|--|-----|-----|-----|
| 1|!2   |false|false|
| 2|!2,!3|false|found|

So `exact` should be true when the first `!` found.

```go
if !idr.bang {
	exact = true
	if idr.lo <= eventID && eventID <= idr.hi {
		found = true
	}
} else {
	if idr.lo <= eventID && eventID <= idr.hi {
		found = false
		exact = true
	} else if !exact {
		found = true
	}
}
```

I'll add test for this edge case. Thanks.

**Comment by mattn** [comment]
At all, I notice this `--event-id` is too difficult to use. And there are more edge case. ex: `2,!3,3`. This will make user confusing. So I suggest to change this like `--source-pattern` and `--source-exclude`.

`--event-id-pattern` and `--event-id-exclude` both allow `1,2` or `3-4`. (don't accept `!`)
And looks IDs following.

1. handle `--event-id-pattern`
2. handle `--event-id-exclude`
    Accepted IDs will be canced

How do you think?

**Comment by astj** [comment]
I saw #137 , and it seems nice to me. How do you think? > @daiksy

**Comment by daiksy** [comment]
@astj 
I think that it is a good thought !
